src/features/walletconnect/components/WalletConnectProvider/index.tsx

```diff
@@ -90,9 +88,9 @@ export const WalletConnectProvider = ({ children }: { children: ReactNode }) =>
 
         // Get response from Safe Wallet Provider
         return safeWalletProvider.request(event.id, event.params.request, {
-          id: WalletConnectSafeApp.id,
-          url: WalletConnectSafeApp.url,
-          name: getPeerName(session.peer) || 'Unknown dApp',
+          id: -1,
+          url: session.peer.metadata.url,
+          name: getPeerName(session.peer) || 'WalletConnect',
```

Comment:
Afaiu we are sending this through WC and not proposing it with the transaction to our backend. How does this affect what we display in the tx history?

Reply:
No, this is what we pass to our tx modal. It will use the icon URL and name in the title of the modal. Since the URL won't be a known Safe App (unless the main dapp and a safe app share the same URL), CGW will just ignore it.